client/src/views/RepoViews.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from "react";
+import { http } from "../api/http";
+
+import styled from "styled-components";
+import logo from "../logo.svg";
+import PanelContainer from "../components/PanelContainer";
+
+function RepoViews() {
+  const [content, setData] = useState([]);
+  const [author, setAuthor] = useState("");
+  const [isLoading, setLoading] = useState(false);
+
+  async function fetchData(author) {
+    const result = await http.get(`/api/${author}`);
+    setData(result.data);
+    setLoading(false);
+  }
+
+  function onAuthorSubmit(e) {
+    if (e.charCode === 13) {
+      fetchData(author);
+    }
+  }
+
+  function onRepoSubmit() {
+    setLoading(true);
+    fetchData(author);
+  }
+
+  function renderUserInput() {
+    return (
+      <RepoInput>
+        <input
+          value={author}
+          onChange={e => {
+            setAuthor(e.target.value);
+          }}
+          onKeyPress={e => onAuthorSubmit(e)}
+        />
+        <button onClick={onRepoSubmit} disabled={!author}>
+          GO
+        </button>
+      </RepoInput>
+    );
+  }
+
+  return (
+    <div>
+      {renderUserInput()}
+      {isLoading ? (
+        <img src={logo} className="App-logo" alt="logo" />
+      ) : (
+        <PanelContainer content={content} />
+      )}
+    </div>
+  );
+}
+
+const RepoInput = styled.div`
```

Comment:
I'm curious, why are we using styles like this instead of having them in a `.(s)css` file?

Reply:
https://blog.bitsrc.io/5-ways-to-style-react-components-in-2019-30f1ccc2b5b

Reply:
Here is an interesting read for you:
https://blog.bitsrc.io/5-ways-to-style-react-components-in-2019-30f1ccc2b5b